### PR TITLE
Added formula layer parameter to geom_smooth

### LINKED ISSUE #4330: Adding geom_lm and geom_spline and more geoms?
The package **ggformula** contains a geom (**geom_lm**) that (I think) answers a problem I have had, namely to be able to fit parallel lines to a linear model plot.  It also has **geom_spline** that would be useful.

It also references many other "standard" geoms from ggplot2 that we don't yet include.  Some sound useful and I wonder why not?  This is not urgent, but is there any reason (in particular for the general plotting) that we omit geoms that are in the ggplot2 package?  

I realise there are many other tasks in R-Instat, but is other work the only reason?

Incidentally the mosaic project used to use the lattice package for the graphics, but it seems to have migrated recently to ggplot - now allowing both.  They seem to have written the ggformula package partly to have a similar command structure for ggplot as for the lattic graphics system  They also have this simple vignette that compares the graphs for the 2 systems 
[https://cran.r-project.org/web/packages/mosaic/vignettes/ggformula-lattice.pdf](url)

### PR DESCRIPTION
**Pull request by Ogik99**
fixes #4330 

I have added a number of formulas to geom_smooth guided by the following reference:
https://ggplot2.tidyverse.org/reference/geom_smooth.html
With the formula options we are able to have geom_spline.

geom_tufteboxplot is still under progress.

### CODE REVIEW

**Comment by dannyparsons** [comment]
Do we already have the `splines` package included? If not we'll need to add it.

**Comment by maxwellfundi** [comment]
@Ogik99  Is the geom_tufte ready now?

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -1266,7 +1267,7 @@ Public Class ucrGeom
         'Can we have  stack, DoDragDrop, fill positions here)
         clsgeom_smooth.AddLayerParameter("method", "list", Chr(34) & "lm" & Chr(34), lstParameterStrings:={Chr(34) & "lm" & Chr(34), Chr(34) & "glm" & Chr(34), Chr(34) & "gam" & Chr(34), Chr(34) & "loess" & Chr(34), Chr(34) & "rlm" & Chr(34)})
         'formula has to be an input and we dont have that currently. its passed in like this formula= y ~ x or  formula= y ~ poly(x, 2) or formula= y ~ log(x) so the user has to type in stuff
-        'clsgeom_smooth.AddLayerParameter("formula",)
+        clsgeom_smooth.AddLayerParameter("formula", "list", Chr(34) & "y ~ x" & Chr(34), lstParameterStrings:={Chr(34) & "y ~ x" & Chr(34), Chr(34) & "y ~ poly(x, 2)" & Chr(34), Chr(34) & "y ~ log(x)" & Chr(34), Chr(34) & " y ~ splines::bs(x,3)" & Chr(34)})
```
Comment (maxwellfundi) [inline instat/ucrGeom.vb]: @Ogik99  Am just curious, can this be anything the user wants? Can we make this editable @dannyparsons if these are some of the formulas, then I think the user might want some freedom to type their own formula?

**Comment by rdstern** [comment]
From what I understand, making it editable would be great.

File: instat/ucrGeom.vb (hunk 1 above)
Comment (dannyparsons) [reply instat/ucrGeom.vb]: Are the list types non editable at the moment? These are nice options but we do need to be able to type anything. I thought you added an option to have an editable or non editable list for these types?

Comment (maxwellfundi) [reply instat/ucrGeom.vb]: Yes i did- So we now have `list` which is not editable and what I added `editablelist`. @Ogik99 please change this from `list `to `editablelist`

**Comment by Ogik99** [comment]
> @Ogik99 Is the geom_tufte ready now?

geom_tufte is now ready

File: instat/ucrGeom.vb (hunk 1 above)
Comment (Ogik99) [reply instat/ucrGeom.vb]: > Yes i did- So we now have `list` which is not editable and what I added `editablelist`. @Ogik99 please change this from `list `to `editablelist`

This has been changed.

**Comment by Ogik99** [comment]
> Do we already have the `splines` package included? If not we'll need to add it.

currently work on the 'splines' package

**Comment by maxwellfundi** [comment]
@Ogik99 i think @dannyparsons  mentioned that the `splines` is part of the base packages and needs not to be added #5376

File: instat/ucrGeom.vb [hunk 2]
```diff
@@ -1399,6 +1400,43 @@ Public Class ucrGeom
...
+        clsgeom_tufteboxplot.AddAesParameter("size", strIncludedDataTypes:={"factor", "numeric"})
+        clsgeom_tufteboxplot.AddAesParameter("linetype", strIncludedDataTypes:={"factor"})
+        clsgeom_tufteboxplot.AddAesParameter("shape", strIncludedDataTypes:={"factor"})
+        clsgeom_tufteboxplot.AddAesParameter("fill", strIncludedDataTypes:={"factor", "numeric"})
+        clsgeom_tufteboxplot.AddAesParameter("alpha", strIncludedDataTypes:={"factor", "numeric"})
+
+        'Layer parameters
+        clsgeom_tufteboxplot.AddLayerParameter("stat", "list", Chr(34) & "fivenumber" & Chr(34))
+        clsgeom_tufteboxplot.AddLayerParameter("position", "list", Chr(34) & "dodge" & Chr(34))
+        clsgeom_tufteboxplot.AddLayerParameter("outlier.colour", "colour", Chr(34) & "black" & Chr(34))
```
Comment (maxwellfundi) [inline instat/ucrGeom.vb]: @Ogik99 This needs default values else they will have developer errors.

Comment (maxwellfundi) [reply instat/ucrGeom.vb]: Everything else looks okay.

**Comment by Wycklife** [comment]
@maxwellfundi This seems okay from my end

File: instat/ucrGeom.vb (hunk 2 above)
Comment (Ogik99) [reply instat/ucrGeom.vb]: I have added the default values

Comment (maxwellfundi) [reply instat/ucrGeom.vb]: Great! Looks good,

**Pull request merged**